manager/controllers/app/fybrikapplication_controller.go

```diff
@@ -412,6 +418,32 @@ func CreateDataRequest(application *api.FybrikApplication, dataCtx api.DataConte
 	}
 }
 
+func (r *FybrikApplicationReconciler) ValidateAssetResponse(response *datacatalog.GetAssetResponse, taxonomyFile string) error {
+	var allErrs []*field.Error
+
+	// Convert GetAssetRequest Go struct to JSON
+	responseJSON, err := json.Marshal(response)
+	if err != nil {
+		return err
+	}
+	r.Log.Info().Msg("responseJSON:" + string(responseJSON))
+
+	// Validate Fybrik module against taxonomy
+	allErrs, err = validate.TaxonomyCheck(responseJSON, taxonomyFile)
+	if err != nil {
+		return err
+	}
+
+	// Return any error
+	if len(allErrs) == 0 {
+		return nil
+	}
+
+	return apierrors.NewInvalid(
+		schema.GroupKind{Group: "app.fybrik.io", Kind: "DataCatalog-AssetResponse"},
+		response.ResourceMetadata.Name, allErrs)
```

Comment:
ResourceMetadata.Name is optional, better use dataset id

Reply:
Actually dataset id is stored as response.ResourceMetadata.Name in the response object. There is no other field in the response object which explicitly stores the dataset id. Hope this is ok ?

Reply:
You could pass req.Context.DataSetID to ValidateAssetResponse and use that.
ResourceMetadata.Name is optional, thus we don't want to rely on it.

Reply:
sure will do.